diff --git a/include/env_default.h b/include/env_default.h
index a2ccb4b7c9..699f0d6f23 100644
--- a/include/env_default.h
+++ b/include/env_default.h
@@ -37,8 +37,11 @@ const uchar default_environment[] = {
 #ifdef	CONFIG_USE_BOOTARGS
 	"bootargs="	CONFIG_BOOTARGS			"\0"
 #endif
-#ifdef	CONFIG_BOOTCOMMAND
-	"bootcmd="	CONFIG_BOOTCOMMAND		"\0"
+#ifdef	CONFIG_MENDER_BOOTCOMMAND
+	"bootcmd=if test ${boot_device} = serial || "
+	"test ${boot_device} = usb;then "
+	"stm32prog ${boot_device} ${boot_instance}; "
+	"else "	CONFIG_MENDER_BOOTCOMMAND	"; fi\0"
 #endif
 #ifdef	CONFIG_RAMBOOTCOMMAND
 	"ramboot="	CONFIG_RAMBOOTCOMMAND		"\0"
