From a0c9719860f0fe4a51a79186efc8a8bf3d4dc7a8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ant=C3=B3nio=20Oliveira?= <antonio.oliveira@konsulko.com>
Date: Mon, 16 Jan 2023 15:59:43 +0000
Subject: [PATCH] disable config-is-nowhere sanity check

---
 include/config_mender.h | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/include/config_mender.h b/include/config_mender.h
index 3377ab0a1b..4b31ef5ad1 100644
--- a/include/config_mender.h
+++ b/include/config_mender.h
@@ -24,9 +24,15 @@
 
 #include <config_mender_defines.h>
 
-#ifdef CONFIG_ENV_IS_NOWHERE
-# error A CONFIG_ENV_IS_IN_<storage-type> define is required for Mender to work. For standard Mender setups this should be CONFIG_ENV_IS_IN_MMC for HD/SSD/MMC/SD storage setup, and CONFIG_ENV_IS_IN_UBI for Flash/UBI storage.
-#endif
+/*
+ * u-boot-stm32mp **does need** CONFIG_ENV_IS_NOWHERE
+ * to support operation from DFU (usb or uart)
+ * Disabling this config causes u-boot to silently hang
+ * when running from DFU, which is a nasty thing to track down.
+ */
+//#ifdef CONFIG_ENV_IS_NOWHERE
+//# error A CONFIG_ENV_IS_IN_<storage-type> define is required for Mender to work. For standard Mender setups this should be CONFIG_ENV_IS_IN_MMC for HD/SSD/MMC/SD storage setup, and CONFIG_ENV_IS_IN_UBI for Flash/UBI storage.
+//#endif
 
 #ifndef CONFIG_BOOTCOUNT_LIMIT
 # error CONFIG_BOOTCOUNT_LIMIT is required for Mender to work. Make sure that: 1) All the instructions at https://docs.mender.io/system-updates-yocto-project/board-integration/bootloader-support/u-boot have been followed. 2) All required layers are included in bblayers.conf, including any board specific layers such as meta-mender-<board>
