diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index c576d65efa..bc5b76172c 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -259,6 +259,12 @@
 			"run bootcmd_mmc1; "				\
 		"fi\0"

+/*
+ * Mender
+ */
+#define CONFIG_BOOTCOUNT_LIMIT
+#define CONFIG_BOOTCOUNT_ENV
+
 #define BOOTENV_DEV_NAME_MMC_AUTO(devtypeu, devtypel, instance) \
 	"mmc_auto "

