From 55a4ddff68fbca30333ff2ceea77f247caf44493 Mon Sep 17 00:00:00 2001
From: Marek Belisko <marek.belisko@open-nandra.com>
Date: Wed, 5 Aug 2020 20:59:01 +0200
Subject: [PATCH] configs: sunxi: add Mender required options

Signed-off-by: Mirza Krak <mirza.krak@northern.tech>
Signed-off-by: Marek Belisko <marek.belisko@open-nandra.com>
---
 include/configs/sunxi-common.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 7260eb72a4..0ad7cabc3a 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -99,6 +99,10 @@
 
 #if defined(CONFIG_ENV_IS_IN_MMC)
 
+#define CONFIG_BOOTCOUNT_LIMIT
+#define CONFIG_BOOTCOUNT_ENV
+#define CONFIG_SYS_REDUNDAND_ENVIRONMENT
+
 #ifdef CONFIG_ARM64
 /*
  * This is actually (CONFIG_ENV_OFFSET -
-- 
2.25.1

