From dc1fb0732d163542b60ed8023f2c44dcb67193f0 Mon Sep 17 00:00:00 2001
From: Mirza Krak <mirza.krak@northern.tech>
Date: Mon, 6 Aug 2018 21:32:09 +0200
Subject: [PATCH 2/2] configs: sunxi: add Mender required options

Signed-off-by: Mirza Krak <mirza.krak@northern.tech>
---
 include/configs/sunxi-common.h | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index e4e7c22778..611980b603 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -142,6 +142,9 @@
 
 #if defined(CONFIG_ENV_IS_IN_MMC)
 
+#define CONFIG_BOOTCOUNT_LIMIT
+#define CONFIG_BOOTCOUNT_ENV
+
 #ifdef CONFIG_ARM64
 /*
  * This is actually (CONFIG_ENV_OFFSET -
@@ -151,16 +154,8 @@
 #define CONFIG_BOARD_SIZE_LIMIT		0x7e000
 #endif
 
-#if CONFIG_MMC_SUNXI_SLOT_EXTRA != -1
-/* If we have two devices (most likely eMMC + MMC), favour the eMMC */
-#define CONFIG_SYS_MMC_ENV_DEV		1
-#else
-/* Otherwise, use the only device we have */
-#define CONFIG_SYS_MMC_ENV_DEV		0
-#endif
 #define CONFIG_SYS_MMC_MAX_DEVICE	4
-#elif defined(CONFIG_ENV_IS_NOWHERE)
-#define CONFIG_ENV_SIZE			(128 << 10)
+
 #endif
 
 #ifndef CONFIG_MACH_SUN8I_V3S
-- 
2.18.0

