From 01df6062dff7240fef99c77f3a6a8dcb2f0f0521 Mon Sep 17 00:00:00 2001
From: Marek Belisko <marek.belisko@open-nandra.com>
Date: Mon, 24 May 2021 10:30:36 +0200
Subject: [PATCH] sunxi-common: Increase SIZE_LIMIT to fit with mender
 extensions

Signed-off-by: Marek Belisko <marek.belisko@open-nandra.com>
---
 include/configs/sunxi-common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 0ef289f..1482fe6 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -131,7 +131,7 @@
  * (CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_SECTOR * 512)), but the value will be used
  * directly in a makefile, without the preprocessor expansion.
  */
-#define CONFIG_BOARD_SIZE_LIMIT		0x7e000
+#define CONFIG_BOARD_SIZE_LIMIT		0xaffff
 #endif
 
 #if CONFIG_MMC_SUNXI_SLOT_EXTRA != -1
-- 
2.7.4

