diff --git a/meta-mender-qemu/conf/layer.conf b/meta-mender-qemu/conf/layer.conf
index 1811af6..f4fd396 100644
--- a/meta-mender-qemu/conf/layer.conf
+++ b/meta-mender-qemu/conf/layer.conf
@@ -12,7 +12,7 @@ BBFILE_COLLECTIONS += "mender-qemu"
 BBFILE_PATTERN_mender-qemu = "^${LAYERDIR}/"
 BBFILE_PRIORITY_mender-qemu = "6"
 
-LAYERSERIES_COMPAT_mender-qemu = "scarthgap"
+LAYERSERIES_COMPAT_mender-qemu = "scarthgap wrynose"
 LAYERDEPENDS_mender-qemu = "mender"
 
 MENDER_EFI_LOADER:mender-image-uefi:qemux86 = "ovmf"
