From 0ca8cf728807766f9bc36fe8551b96f3295273b5 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Fri, 10 Jan 2020 12:19:14 -0800
Subject: [PATCH] Move jetson-nano env to SDcard

---
 include/configs/p3450-porg.h | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/include/configs/p3450-porg.h b/include/configs/p3450-porg.h
index c3e709d44d..89ce624a8c 100644
--- a/include/configs/p3450-porg.h
+++ b/include/configs/p3450-porg.h
@@ -47,18 +47,14 @@
 	func(PXE, pxe, na) \
 	func(DHCP, dhcp, na)
 
-/* Environment at end of QSPI, in the VER partition */
-#define CONFIG_ENV_IS_IN_SPI_FLASH
 #define CONFIG_ENV_SPI_MAX_HZ		48000000
 #define CONFIG_ENV_SPI_MODE		SPI_MODE_0
-#define CONFIG_ENV_SECT_SIZE		SZ_4K
-/* 64KB at end of SPI flash is used by NVIDIA, and we need 256K to
-   hold 2*128K env blocks, so position at 320K (256K + 64K) from end,
-   in the padding area.  */
-#define CONFIG_ENV_OFFSET		(SZ_4M - 327680)
-/* ONLY for SPI flash, this does NOT get defined automatically
-   if CONFIG_ENV_xxx_REDUND settings are present. */
-#define CONFIG_SYS_REDUNDAND_ENVIRONMENT
+/* Mender: use SDcard instead of SPI flash for environment */
+#define CONFIG_ENV_IS_IN_MMC		1
+#define CONFIG_SYS_MMC_ENV_DEV		1
+#define CONFIG_SYS_MMC_ENV_PART		0
+#define CONFIG_ENV_OFFSET (MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_1)
+#define CONFIG_ENV_OFFSET_REDUND (MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_2)
 
 /* SPI */
 #define CONFIG_SF_DEFAULT_MODE		SPI_MODE_0
-- 
2.25.1

