diff --git a/include/configs/p3450-porg.h b/include/configs/p3450-porg.h
index 7be05f3dae..2e97052610 100644
--- a/include/configs/p3450-porg.h
+++ b/include/configs/p3450-porg.h
@@ -35,7 +35,7 @@
 	func(DHCP, dhcp, na)
 
 /* Environment s/b at end of SPI, fix it later */
-#define CONFIG_ENV_IS_IN_SPI_FLASH
+#define CONFIG_ENV_IS_IN_MMC
 /* 64KB at end of SPI flash is used, so position at 128K from end
    in the padding area */
 #define CONFIG_ENV_OFFSET		(CONFIG_SPI_FLASH_SIZE-131072)
