diff -r -u /workdir/var-som-mx8m-mini-dunfell-no-mender/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/old-src/configs/imx8mm_var_dart_defconfig /workdir/var-som-mx8m-mini-dunfell-no-mender/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/configs/imx8mm_var_dart_defconfig
--- a/configs/imx8mm_var_dart_defconfig	2023-02-24 04:38:53.145452295 +0000
+++ b/configs/imx8mm_var_dart_defconfig	2023-02-24 04:39:33.986531500 +0000
@@ -10,8 +10,6 @@
 CONFIG_SYS_I2C_MXC_I2C2=y
 CONFIG_SYS_I2C_MXC_I2C3=y
 CONFIG_SYS_I2C_MXC_I2C4=y
-CONFIG_ENV_SIZE=0x1000
-CONFIG_ENV_OFFSET=0x400000
 CONFIG_DM_GPIO=y
 CONFIG_TARGET_IMX8MM_VAR_DART=y
 CONFIG_SPL_MMC_SUPPORT=y
@@ -59,7 +57,6 @@
 CONFIG_CMD_FS_GENERIC=y
 CONFIG_OF_CONTROL=y
 CONFIG_DEFAULT_DEVICE_TREE="imx8mm-var-dart-customboard"
-CONFIG_ENV_IS_IN_MMC=y
 CONFIG_SYS_RELOC_GD_ENV_ADDR=y
 CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
 CONFIG_NET_RANDOM_ETHADDR=y
@@ -109,3 +106,32 @@
 CONFIG_CI_UDC=y
 CONFIG_SDP_LOADADDR=0x40400000
 CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_ENV_SIZE=0x1000
+CONFIG_ENV_OFFSET=0x400000
+CONFIG_ENV_OFFSET_REDUND=0x800000
+CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
+# CONFIG_ENV_IS_NOWHERE is not set
+# CONFIG_ENV_IS_IN_EEPROM is not set
+# CONFIG_ENV_IS_IN_FAT is not set
+# CONFIG_ENV_IS_IN_EXT4 is not set
+# CONFIG_ENV_IS_IN_FLASH is not set
+# CONFIG_ENV_IS_IN_NAND is not set
+# CONFIG_ENV_IS_IN_NVRAM is not set
+# CONFIG_ENV_IS_IN_ONENAND is not set
+# CONFIG_ENV_IS_IN_REMOTE is not set
+# CONFIG_ENV_IS_IN_SATA is not set
+# CONFIG_ENV_IS_IN_SPI_FLASH is not set
+# CONFIG_ENV_IS_IN_UBI is not set
+CONFIG_MMC=y
+CONFIG_ENV_IS_IN_MMC=y
+# CONFIG_BOOTCOUNT_BOOTLIMIT is not set
+# CONFIG_BOOTCOUNT_GENERIC is not set
+# CONFIG_BOOTCOUNT_EXT is not set
+# CONFIG_BOOTCOUNT_AM33XX is not set
+# CONFIG_BOOTCOUNT_RAM is not set
+# CONFIG_BOOTCOUNT_I2C is not set
+# CONFIG_BOOTCOUNT_AT91 is not set
+# CONFIG_BOOTCOUNT_ALEN is not set
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
+# CONFIG_BOOTCOMMAND is not set
Binary files /workdir/var-som-mx8m-mini-dunfell-no-mender/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/old-src/.git/index and /workdir/var-som-mx8m-mini-dunfell-no-mender/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/.git/index differ
diff -r -u /workdir/var-som-mx8m-mini-dunfell-no-mender/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/old-src/include/config_defaults.h /workdir/var-som-mx8m-mini-dunfell-no-mender/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/include/config_defaults.h
--- a/include/config_defaults.h	2023-02-24 04:38:52.848451719 +0000
+++ b/include/config_defaults.h	2023-02-24 04:39:33.527530610 +0000
@@ -10,3 +10,5 @@
 #define _CONFIG_DEFAULTS_H_
 
 #endif
+#define CONFIG_SYS_MMC_ENV_DEV 0
+#define CONFIG_SYS_MMC_ENV_PART 0
diff -r -u /workdir/var-som-mx8m-mini-dunfell-no-mender/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/old-src/include/configs/imx8mm_var_dart.h /workdir/var-som-mx8m-mini-dunfell-no-mender/build/tmp/work/imx8mm_var_dart-fslc-linux/u-boot-variscite/1.0-r0/git/include/configs/imx8mm_var_dart.h
--- a/include/configs/imx8mm_var_dart.h	2023-02-24 04:38:52.858451739 +0000
+++ b/include/configs/imx8mm_var_dart.h	2023-02-24 04:39:33.534530624 +0000
@@ -69,7 +69,7 @@
 	"image=Image.gz\0" \
 	"console=undefined\0" \
 	"img_addr=0x42000000\0"			\
-	"fdt_addr=0x43000000\0"			\
+	"fdt_addr_r=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fdt=try\0" \
 	"ip_dyn=yes\0" \
@@ -101,7 +101,7 @@
 			"fi; " \
 		"fi; \0" \
 	"mmcargs=run setconsole; setenv bootargs console=${console} " \
-		"root=/dev/mmcblk${mmcblk}p${mmcpart} rootwait rw ${cma_size}\0 " \
+		"rootwait rw ${cma_size}\0 " \
 	"loadbootscript=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${bootdir}/${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
@@ -145,7 +145,7 @@
 			"echo wait for boot; " \
 		"fi;\0" \
 	"netargs=run setconsole; setenv bootargs console=${console} " \
-		"root=/dev/nfs ${cma_size} " \
+		"${cma_size} " \
 		"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0" \
 	"netboot=echo Booting from net ...; " \
 		"if test ${ip_dyn} = yes; then " \
@@ -168,25 +168,6 @@
 			"booti; " \
 		"fi;\0"
 
-#define CONFIG_BOOTCOMMAND \
-	"run ramsize_check; " \
-	"mmc dev ${mmcdev}; "\
-	"if mmc rescan; then " \
-		"if test ${use_m4} = yes && run loadm4bin; then " \
-			"run runm4bin; " \
-		"fi; " \
-		"if run loadbootscript; then " \
-			"run bootscript; " \
-		"else " \
-			"if run loadimage; then " \
-				"run mmcboot; " \
-			"else " \
-				"run netboot; " \
-			"fi; " \
-		"fi; " \
-	"else " \
-		"booti ${loadaddr} - ${fdt_addr}; " \
-	"fi;"
 
 /* Link Definitions */
 #define CONFIG_LOADADDR			0x40480000
@@ -201,7 +182,6 @@
 	(CONFIG_SYS_INIT_RAM_ADDR + CONFIG_SYS_INIT_SP_OFFSET)
 
 #define CONFIG_ENV_OVERWRITE
-#define CONFIG_SYS_MMC_ENV_DEV		1   /* USDHC2 */
 
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN		SZ_32M
