From e4ba4bbdf5a348c0fd0ffdc9b9fd119d8479143a Mon Sep 17 00:00:00 2001
From: Mirza Krak <mirza.krak@northern.tech>
Date: Sat, 13 Apr 2019 18:35:09 +0200
Subject: [PATCH 1/1] configs: meson64: add Mender requirements

Signed-off-by: Mirza Krak <mirza.krak@northern.tech>
---
 configs/odroid-c2_defconfig | 1 +
 include/configs/meson64.h   | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/configs/odroid-c2_defconfig b/configs/odroid-c2_defconfig
index 33e6e74fa2..79cf159be3 100644
--- a/configs/odroid-c2_defconfig
+++ b/configs/odroid-c2_defconfig
@@ -38,3 +38,4 @@ CONFIG_DEBUG_UART_ANNOUNCE=y
 CONFIG_DEBUG_UART_SKIP_INIT=y
 CONFIG_MESON_SERIAL=y
 CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_ENV_IS_IN_MMC=y
diff --git a/include/configs/meson64.h b/include/configs/meson64.h
index 40ac079dfc..03ad3b6c6f 100644
--- a/include/configs/meson64.h
+++ b/include/configs/meson64.h
@@ -28,6 +28,9 @@
 #define CONFIG_SYS_LOAD_ADDR		CONFIG_SYS_TEXT_BASE
 #define CONFIG_SYS_BOOTM_LEN		(64 << 20) /* 64 MiB */
 
+#define CONFIG_BOOTCOUNT_LIMIT
+#define CONFIG_BOOTCOUNT_ENV
+
 /* ROM USB boot support, auto-execute boot.scr at scriptaddr */
 #define BOOTENV_DEV_ROMUSB(devtypeu, devtypel, instance) \
 	"bootcmd_romusb=" \
-- 
2.23.0

