diff --git edk2-tegra.a/edk2-nvidia/Silicon/NVIDIA/Drivers/BootChainDxe/BootChainDxe.c edk2-tegra.b/edk2-nvidia/Silicon/NVIDIA/Drivers/BootChainDxe/BootChainDxe.c
index cfaf6872..d7eaf0f3 100644
--- edk2-tegra.a/edk2-nvidia/Silicon/NVIDIA/Drivers/BootChainDxe/BootChainDxe.c
+++ edk2-tegra.b/edk2-nvidia/Silicon/NVIDIA/Drivers/BootChainDxe/BootChainDxe.c
@@ -336,8 +336,8 @@ BootChainExecuteUpdate (
         BCStatus = STATUS_ERROR_BOOT_CHAIN_FAILED;
         goto SetStatusAndBootOs;
       }
-    } else {
-      // Status is already ERROR or SUCCESS, finish the update and boot OS
+    } else if (BCStatus != STATUS_SUCCESS) {
+      // Status is already ERROR, finish the update and boot OS
       goto FinishUpdateAndBootOs;
     }
   }
