diff --git edk2-tegra.a/edk2-nvidia/Silicon/NVIDIA/Tegra/DeviceTree/L4TConfiguration.dts edk2-tegra.b/edk2-nvidia/Silicon/NVIDIA/Tegra/DeviceTree/L4TConfiguration.dts
index 2c619e21..b254ae76 100644
--- edk2-tegra.a/edk2-nvidia/Silicon/NVIDIA/Tegra/DeviceTree/L4TConfiguration.dts
+++ edk2-tegra.b/edk2-nvidia/Silicon/NVIDIA/Tegra/DeviceTree/L4TConfiguration.dts
@@ -37,7 +37,7 @@
 								locked;
 							};
 							RootfsRedundancyLevel {
-								data = [00 00 00 00];
+								data = [01 00 00 00];
 								runtime;
 								locked;
 							};
